src/main/java/io/cryostat/core/agent/Field.java

```diff
@@ -0,0 +1,130 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.core.agent;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+public class Field extends CapturedValue {
+
+    private static final String DEFAULT_FIELD_NAME = "New Field"; // $NON-NLS-1$
+    private static final String DEFAULT_FIELD_EXPRESSION = "myField"; // $NON-NLS-1$
+    private static final String EXPRESSION_REGEX =
+            "([a-zA-Z_$][a-zA-Z0-9_$]*\\.)*([a-zA-Z_$][a-zA-Z0-9_$]*)(\\.[a-zA-Z_$][a-zA-Z_$]*)*"; // $NON-NLS-1$
+
+    private static final String XML_TAG_FIELD = "field"; // $NON-NLS-1$
+    private static final String XML_TAG_EXPRESSION = "expression"; // $NON-NLS-1$
+
+    private final Event event;
+
+    private String expression;
+
+    Field(Event event) {
+        super();
+        this.event = event;
+
+        expression = DEFAULT_FIELD_EXPRESSION;
+        setName(DEFAULT_FIELD_NAME);
+    }
+
+    Field(Event event, Element element) {
+        super(element);
+        this.event = event;
+
+        NodeList elements = element.getElementsByTagName(XML_TAG_EXPRESSION);
+        if (elements.getLength() != 0) {
+            expression = elements.item(0).getTextContent();
+        }
+    }
+
+    @Override
+    public Element buildElement(Document document) {
+        Element element = super.buildElement(document);
+        element = (Element) document.renameNode(element, null, XML_TAG_FIELD);
+
+        Element expressionElement = document.createElement(XML_TAG_EXPRESSION);
+        expressionElement.setTextContent(expression);
+        element.appendChild(expressionElement);
+
+        return element;
+    }
+
+    @Override
+    public void setName(String name) {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException("name cannot be empty or null");
+        }
+
+        super.setName(name);
+    }
+
+    public String getExpression() {
+        return expression;
+    }
+
+    public void setExpression(String expression) {
+        if (expression == null || expression.isEmpty()) {
+            throw new IllegalArgumentException("expression cannot be empty or null");
+        }
+
+        expression = expression.trim();
+        if (!expression.matches(EXPRESSION_REGEX)) {
+            throw new IllegalArgumentException("expression has incorrect syntax");
+        }
+
+        this.expression = expression;
+    }
+
+    public Field createWorkingCopy() {
+        Field copy = new Field(event);
+
+        copyContentToWorkingCopy(copy);
+        copy.expression = expression;
+
+        return copy;
+    }
+
+    /*
+    public Field createDuplicate() {
+        Field duplicate = createWorkingCopy();
+        duplicate.setName(event.nextUniqueFieldName(getName()));
+
+        return duplicate;
+    }*/
```

Comment:
Is this commented out method worth keeping?

Reply:
This is not needed, it's been removed